functions-and-operators/json-functions/json-functions-validate.md

```diff
@@ -0,0 +1,377 @@
+| `required`             | `object` | The property names that are required |
+
+Examples:
+
+For some of the examples we will use this JSON document:
+
+```json
+{
+    "fruits": [
+        "orange",
+        "apple",
+        "pear"
+    ],
+    "vegetables": [
+        "carrot",
+        "pepper",
+        "kale"]
+}
+```
+
+We will use a [user defined variable](/user-defined-variables.md) to hold the JSON document.
+
+```sql
+SET @j := '{"fruits": ["orange", "apple", "pear"], "vegetables": ["carrot", "pepper", "kale"]}';
+```
+
+Let's start by testing the type::
+
+```sql
+SELECT JSON_SCHEMA_VALID('{"type": "object"}',@j);
+```
+
+```
++--------------------------------------------+
+| JSON_SCHEMA_VALID('{"type": "object"}',@j) |
++--------------------------------------------+
+|                                          1 |
++--------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+```sql
+SELECT JSON_SCHEMA_VALID('{"type": "array"}',@j);
+```
+
+```
++-------------------------------------------+
+| JSON_SCHEMA_VALID('{"type": "array"}',@j) |
++-------------------------------------------+
+|                                         0 |
++-------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+```sql
+mysql> SELECT JSON_TYPE(@j);
+```
+
+```
++---------------+
+| JSON_TYPE(@j) |
++---------------+
+| OBJECT        |
++---------------+
+1 row in set (0.00 sec)
+```
+
+As you can see in the output above, the type is `object`. This matches with the output of [`JSON_TYPE()`](/functions-and-operators/json-functions/json-functions-return.md#json_type).
+
+Now let's validate the presence of certain attributes.
+
+```sql
+SELECT JSON_SCHEMA_VALID('{"required": ["fruits","vegetables"]}',@j);
+```
+
+```
++---------------------------------------------------------------+
+| JSON_SCHEMA_VALID('{"required": ["fruits","vegetables"]}',@j) |
++---------------------------------------------------------------+
+|                                                             1 |
++---------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+In the output above you can see that see that validation of the presence of the `fruits` and `vegetables` attributes succeeded.
+
+```sql
+SELECT JSON_SCHEMA_VALID('{"required": ["fruits","vegetables","grains"]}',@j);
+```
+
+```
++------------------------------------------------------------------------+
+| JSON_SCHEMA_VALID('{"required": ["fruits","vegetables","grains"]}',@j) |
++------------------------------------------------------------------------+
+|                                                                      0 |
++------------------------------------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+In the output above you can see that see that validation of the presence of the `fruits`, `vegetables` and `grains` attributes failed as the latter isn't present.
```

Comment:
```suggestion
In the preceding output, you can see that see that validation of the presence of the `fruits`, `vegetables` and `grains` attributes fails because the latter is not present.
```